src/ci/shared.sh

```diff
@@ -15,6 +15,12 @@
 
 # See http://unix.stackexchange.com/questions/82598
 function retry {
+  # Only do retries on CI, locally they shouldn't really be necessary
+  if [ -z "${CI:-}" ]; then
+      "$@"
+      return
+  fi
+
```

Comment:
Oops sorry I meant to respond to this earlier! I changed this here because it's a pretty bad experience when working on the docker files. Every failure turns into a "retry" so if you've got a normal failure in the docker image it just gives it to you 5 times which often makes it hard to find the error.

This seems contentious though so I'll just back it out.